apps/storefront/src/pages/QuoteDraft/index.test.tsx

```diff
@@ -367,6 +409,20 @@ const storeInfoWithDateFormat = buildStoreInfoStateWith({ timeFormat: { display:
 
 const preloadedState = { company: approvedB2BCompany, storeInfo: storeInfoWithDateFormat };
 
+const mockedUsedNavigate = vi.fn();
+
+vi.mock('react-router-dom', async () => {
+  const actual = await vi.importActual('react-router-dom');
+  return {
+    ...actual,
+    useNavigate: () => mockedUsedNavigate,
+  };
+});
```

Comment:
`renderWithProviders` returns a `navigation` method that you can use to check if navigation happened. No need to mock this:

```tsx
const { navigation } = renderWithProviders();
```

If you want to check it wasn't called, make sure to clear the mock before performing the action:

```tsx
navigation.clearMock();

await userEvent.click(submitButton);


expect(navigation).not.toHaveBeenCalled();
```